pkg/relayer/miner/miner.go

```diff
@@ -37,7 +40,7 @@ type miner struct {
 // - ProofQueryClient
 //
 // Available options:
-//   - WithDifficulty
+//   - WithRelayDifficultyTargetHash
```

Comment:
We should discard this option and have a dependency on proof params client to fetch difficulty at the beginning of each session.

Reply:
DItto. THe TODO in #699 (which will be merged into this PR once approved) captures this.

I will do it in followup work.